engine/http.go

```diff
@@ -38,5 +40,20 @@ func (eng *Engine) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	// can only be sent before the body.
 	// (we could possibly use http footers via chunked encoding, but I couldn't find
 	// how to use them in net/http)
+
+	// Cancel the job if the HTTP connection is closed.
+	if closeNotifier, ok := w.(http.CloseNotifier); ok {
```

Comment:
Near as I can tell this code is never used - I removed it and everything still seems to work.  Do we really need it?
The reason I even thought of this was because I thought this cancel stuff would work for all http interactions but I didn't so I was investigating.  Would it make sense to make it more generic so job.Cancel() is called for all types of jobs?